changelog/target.md

```diff
@@ -0,0 +1,72 @@
+Add `-target=<triple>` for operating system, c, and c++ runtime cross compilation
+
+The command line switch `-target=<triple>` can be used to specify the target the
+compiler should produce code for. The format for `<triple>` is `<arch>-[<vendor>-]<os>[-<cenv>[-<cppenv]]`. Specific values of the specifiers are 
+shown in the tables below.
+
+`<arch>` 
+---------
+`<arch>` is an Architecture bitness, optionally followed by instruction set (which duplicate the functionality of the `-mcpu` flag)
+| `<arch>`  | Architecture bitness |
+| ---------- |-------------| 
+| `x86_64`  |  64 bit |
+| `x64`        |  64 bit |
+| `x86`        |  32 bit |
+| `x32`        |  64bit with 32 bit pointers |
+
+| subarch features | equivalent `-mcpu` flag |
+| ---------- |-------------| 
+| `+sse2`    | `baseline` |
+| `+avx`      |  `avx` |
+| `+avx2`    |  `avx2` |
+
+The architecture and subarch features are concatenated, so `x86_64+avx2` is 64bit with `avx2` instructions.
+
+
+`<vendor>` 
+---------
+`<vendor>` is always ignored, but supported for easier interoperability with other compilers that report and consume target triples.
+
+`<os>`
+------
+
+`<os>` is the operating system specifier. It may be optionally followed by a version number as is `darwin20.3.0` or `freebsd12`. For Freebsd, this sets the corresponding predefined `version` identifier, e.g. `freebsd12` sets `version(FreeBSD12)`. For other operating systems this is for easier interoperability with other compilers.
+
+| `<os>` | Operating system | 
+| ---------- |-------------| 
+| `freestanding` | No operating system |
+| `darwin`  or `osx` | MacOS, iOS, tvOS | 
```

Comment:
* Clang supports `darwin` and `macos`, which both mean macOS. It does not support `osx`

* Why are `iOS` and `tvOS` included? DMD doesn't support those. I know it's a long shot, but if DMD would support these in the future we're painting ourself into a corner, just like Go did. Go used `darwin` + `x86_64` to mean macOS and `darwin` + `arm64` to mean iOS. Then Apple released Mac with ARM64 and then you have already used the obvious combination. I don't know if they managed to solve that in a backwards compatible way. There are also the iOS simulators which are running x86-64. There's no reason DMD could not support those. How would you target those platforms? There are also many other operating systems that DMD don't support which you have not included.

Reply:
fixed